skycoin/skywire
# Integrate m3 dsmgpty with hypervisor and various fixes.

Did you run `make format && make check`?
> Yes

 Changes (closes #155):
* Fixed `RPCClientDialer` logic to not return on hypervisor disconnect.
* Updated hypervisor config format.
* Added hypervisor /pty/{pk} endpoint.
* Added logic to prepare unix file.
* Improved logging.

How to test this:
* Checkout repo `skywire-services@feature/m3-dmsgpty-hypervisor`
* Uncomment `replace` statements in `go.mod` for `skywire-services` and `skywire-mainnet` repos.
* Run the generic integration ENV.
* Visor A via Hypervisor: http://localhost:8080/pty/031b80cd5773143a39d940dc0710b93dcccc262a85108018a7a95ab9af734f8055
* Visor C via Hypervisor: http://localhost:8080/pty/031b80cd5773143a39d940dc0710b93dcccc262a85108018a7a95ab9af734f8055

Changes (fixes #160, fixes #161, fixes #171):
* `managedTransport.close()` now optionally calls `mt.wg.Wait()` to avoid hangs if called in `mt.Serve()`.
* `transport.Manager` deletes `mTp` entry on `SaveTransport` if `mTp` has already stopped serving.
* Improved various logs.
* Added some comments.

How to test this:
* Checkout repo `skywire-services@feature/m3-dmsgpty-hypervisor`
* Uncomment `replace` statements in `go.mod` for `skywire-services` and `skywire-mainnet` repos.
* Run the generic integration ENV.
* Read further instructions in the respective tickets.

Changes (closes #68):
* mTp.redial exponential backoff.
* mTp.updateStatuses looks at discovery response codes to see if tp is still registered.
